### ISSUE TITLE
Training strategy/parameters

### ISSUE THREAD

**Issue by BennoKrojer** [OSU-NLP-Group/MagicBrush#9]
Hi, thanks for this work and great resource! In the paper you mention: "Following prior work [4], we use a 256 × 256 image resolution and the same training strategies and hyper-parameters." Does that mean you kept their train.yaml file exactly the same? I am currently doing something but with different data and so far after 1-2 epochs it looks like total garbage (noisy colored patched more or less), albeit I used a lower batchsize for now.

Also may I ask how long the full training took?

Thanks,
Benno

**Comment by drogozhang** [comment]
Hi Benno, thanks for your interest in our work and the kind words. 

Except with smaller learning rate, we kept the train.yaml file exactly the same. Also, the training takes about 12 hours.

Best,
Kai

**Comment by RotsteinNoam** [comment]
Hey Kai,

Awesome job on this! 👏 Could you share the full fine-tuning details? I'm interested in things like the learning rate, batch size, number of GPUs used, any gradient accumulation strategies, max norm, etc.

Cheers,
Noam

**Comment by drogozhang** [comment]
Hi Noam,

Thanks for your interest, I uploaded the detailed config for fine-tuning [here](https://github.com/OSU-NLP-Group/MagicBrush/blob/main/finetune.yaml).

I remove the concrete paths like ckpt_path. But the rest hyper parameters should be helpful.

Let me know if you need anything else.

Best,
Kai

**Comment by RotsteinNoam** [comment]
Thanks for the answer!

Two followup questions:
1. In the yaml, the number of epochs is 1000, did you have any specific stopping criteria? was there a validation set?
2. In the yaml the validation resolution is 256 (like the training I guess), what is the resolution used for final evaluations in the paper?

Thanks!
Noam

**Comment by drogozhang** [comment]
1. We manually check the output image qualities on the validation set (we provide that on MagicBrush) for each saved checkpoint.
2. As mentioned in the IP2P paper, although the model is trained with 256 resolution, it can generalize to images with larger resolutions. For final evaluation, we don't resize the images, I think it's 1024x1024 IIRC.

Best,
Kai

**Comment by BennoKrojer** [comment]
Hi, thanks for all the nice responses! Much appreciated.

I have 2 more questions:

1. You said "with smaller learning rate" but the lr stayed the same between pix2pix and your yaml file right?
2. I am currently reproducing MagicBrush. Your recommended ckpt is MagicBrush-epoch-52-step-4999, however currently my current ckpt is saved as "epoch=000021-step=000002999" so it would reach step=4999 way before reaching epoch=52, at around epoch=35 already? I am training on 8 GPUs with batchsize 8 since you specified 2 GPUs with batchsize 64, so 8*8. Is there anything I am missing.
Thanks,
Benno

**Comment by drogozhang** [comment]
Hi Benno,

Thank you so much for bringing this up! I double checked the fine-tune yaml and I realized the one with same lr is actually wrong: the last time for this project I was trying to train an editing model from scratch (stable diffusion) with MagicBrush data only, so the learning rate is the same with pix2pix.

I have updated the finetune.yaml and please give it another try with the new one (I fixed the learning rate). 

Thanks again for being so attentive and please let me know if you have any other questions :)

**Comment by BennoKrojer** [comment]
Thanks Kai!
Do you know how you would've gotten the epoch-52-step-4999? What was the number of GPUs, batchsize and grad_accumulation? I train on roughly 8.8K examples, with batchsize 8*8 (8 GPUs and batchsize=8), no grad_accumulation but I get a checkpoint saying epoch=000021-step=000002999. So it would never reach epoch=52 and step=5000

I am just trying to make sure my ablations and in my paper will be correct (and get me the same performance)!

**Comment by BennoKrojer** [comment]
It now says batch_size: 16 in the new yaml, what number of GPUs did you use?

**Comment by drogozhang** [comment]
Sry I may need some time to figure out the details. It was 9 months ago and I can not remember all the details :(

**Comment by BennoKrojer** [comment]
All good! I've been there as well, looking through old files and logs haha. It might be possible to deduce it just by doing the math but I thought I'd rather check with you.

**Comment by drogozhang** [comment]
I think we used 2 gpus. Hope it helps!

**Comment by BennoKrojer** [comment]
https://chat.openai.com/share/ef9998f6-6d2a-4643-8e08-97fc4ecaf862

To match the released checkpoint, I'd have to train with batchsize 43 which is probably not what you had. Is there any chance you had more than ~8.8K datapoints available for training?

If we can't figure it out, I might just retrain myself to have a fair comparison and not compare with paper numbers, that works too I guess!

**Comment by drogozhang** [comment]
There can be <10 examples more during training. Later we found the masks may not be qualified so we deleted them. But I don't think that matters to the training.

I feel it might be a good idea to test the model trained with similar steps. Maybe you can get even better models lol.

**Comment by BennoKrojer** [comment]
Thanks! I will also aim for 5000 steps then with the same learning rate.
Also gonna tweak some things, i.e. curious if training on res=512 improves results.
Did you find the automatic metrics to correlate well with your own eye judgement?

**Comment by BennoKrojer** [comment]
One small quirk of the training I found is that there is a 50% chance the image is flipped, meaning there is no change to learn left or right. In MagicBrush there is not many examples of that but in my additional data a lot.
